### ISSUE TITLE
Snowflake Driver: Allow Using a Private Key Pair To Connect

### ISSUE THREAD

**Issue by spsoto** [TablePlus/TablePlus#2269]
**Please don't confuse with SSH connectivity.** [SnowSQL allows connecting via a private key pair](https://docs.snowflake.com/en/user-guide/snowsql-start.html#private-key-path) instead of a password and an authenticator. 

Unfortunately, this is only vetted way I have to connect to this specific Snowflake account, so I am unable to connect to Snowflake via TablePlus until the option is given. My understanding is that this option is somehow available in some Snowflake drivers as well.

1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):

Snowflake 4.43.1

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):

Build 354

3. The steps to reproduce this issue:

None. The needed option is not there.

**Comment by brunocostalopes** [comment]
I'm also interested in this feature. Any plans for implementing?

**Comment by bertoost** [comment]
This would be great to see implemented

**Comment by furkantektas** [comment]
Indeed, this would be such a useful feature. Since Snowflake will require MFA by April 2025, more people would need MFA auth in the coming months.

**Comment by xantiagoma** [comment]
👀  Please implement this.

**Comment by imakeinternet** [comment]
+1 pretty please
